src/services/catalog.service.ts

```diff
@@ -70,6 +70,30 @@ export class CatalogService { static $inject = ['$filter', '$q', 'Constants', 'A
     return deferred.promise;
   }
 
+  // Get service plans for a service class, takes either the serviceClass object or it's name
+  public getServicePlansForServiceClass(serviceClass: any) : angular.IPromise < any > {
+    let plansVersion = this.apiService.getPreferredVersion('clusterserviceplans');
+    var opts: any = undefined;
+    let serviceClassName = _.get(serviceClass, 'metadata.name', serviceClass);
+
+    // Only request service plans if the resource is available.
+    if (serviceClassName) {
```

Comment:
I think it's better to return null here than return all. I'm worried the dialog might show all plans for all service classes if we accidentally pass a null service class.